source/common/upstream/subset_lb.cc

```diff
@@ -421,31 +426,18 @@ void SubsetLoadBalancer::update(uint32_t priority, const HostVector& hosts_added
   processSubsets(
       hosts_added, hosts_removed,
       [&](LbSubsetEntryPtr entry) {
-        const bool active_before = entry->active();
         entry->priority_subset_->update(priority, hosts_added, hosts_removed);
-
-        if (active_before && !entry->active()) {
-          stats_.lb_subsets_active_.dec();
-          stats_.lb_subsets_removed_.inc();
```

Comment:
I think that the case where an updated entry is now empty (and therefore not active) but still has children doesn't get accounted for anymore? In the purge function, the entry isn't purged (because children) and so the active/removed stats aren't updated.

Reply:
I don't think so... the only way it could be !active (empty) _and_ have children is if it's not initialized, which I think only happens in the `UpdateRemovingUnknownHost` test case.  If it the entry isn't initialized, stats weren't incremented for it in the first place. So we should be good.

Or are you seeing a legit case besides the above?

Reply:
No, just looked that way by inspection.

I suppose if we have a=x → b=y and the only host with a=x is removed, it must also have been the only host with a=x, b=y and therefore that child subset will be purged and then the parent one and the accounting is correct. And if the subset keys were defined such that a=x wasn't an available subset it would never have had any hosts and could never have been active.

I still feel like this should get covered in a test (and it may be).

Reply:
Yep -- I'll follow-up with a few tests to exercise some of these paths.

Reply:
Done.